src/Stratis.Features.Collateral/CollateralFeature.cs

```diff
@@ -38,8 +38,13 @@ public override void Dispose()
     /// </summary>
     public static class FullNodeBuilderCollateralFeatureExtension
     {
+        // Both Cirrus Peg and Cirrus Miner calls this.
         public static IFullNodeBuilder CheckForPoAMembersCollateral(this IFullNodeBuilder fullNodeBuilder)
         {
+            // Inject the CheckCollateralFullValidationRule as the first Full Validation Rule.
```

Comment:
@codingupastorm After a lot of to and fro-ing, this was the only way I could register this rule with doing other hacks in the collateral feature to check whether or not the side chain node was mining (which is when this rule is applicable).

Reply:
Also, is it OK that we execute this rule first like this? I suppose we want to catch collateral errors first?

Reply:
I'm fine with this for now yeah, I like it even. I don't know if it has to be first, but in terms of stopping DoS's maybe that makes the most sense.

I believe that it doesn't depend on anything and nothing depends on it, so it doesn't matter where it goes really though.